nutrition_labels/prodigy_training_data.py

```diff
@@ -10,45 +10,49 @@
 
 import pandas as pd
 
-if __name__ == '__main__':
-    
-    parser = ArgumentParser()
-    parser.add_argument(
-        '--prodigy_data_dir',
-        help='Path to prodigy output dataset file',
-        default='data/prodigy/tech_grants/tech_grants.jsonl'
-    )
-    args = parser.parse_args()
 
-    datestamp = datetime.now().date().strftime('%y%m%d')
+def load_prodigy_tags(prodigy_data_dir):
 
-    cat2bin = {'Not tech grant': 0, 'Tech grant': 1}
+    cat2bin = {"Not tech grant": 0, "Tech grant": 1}
 
     training_data = []
-    with open(args.prodigy_data_dir, 'r') as json_file:
+    with open(prodigy_data_dir, "r") as json_file:
         for json_str in list(json_file):
             data = json.loads(json_str)
-            if data['answer'] != 'ignore':
-                annotation = {
-                    'Internal ID': data['id'],
-                    'Grant texts': data['text']
-                    }
-                label = cat2bin[data['label']]
-                if data['answer']=='accept':
-                    annotation['Relevance code'] = label
+            if data["answer"] != "ignore":
+                annotation = {"Internal ID": data["id"], "Grant texts": data["text"]}
+                label = cat2bin[data["label"]]
+                if data["answer"] == "accept":
+                    annotation["Relevance code"] = label
                 else:
-                    # If label=1, append 0 
+                    # If label=1, append 0
                     # if label=0, append 1
-                    annotation['Relevance code'] = abs(label - 1)
+                    annotation["Relevance code"] = abs(label - 1)
                 training_data.append(annotation)
 
+    return training_data
+
+
+if __name__ == "__main__":
+
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--prodigy_data_dir",
+        help="Path to prodigy output dataset file",
+        default="data/prodigy/tech_grants/tech_grants.jsonl",
+    )
+    args = parser.parse_args()
```

Comment:
I find that a useful pattern is to read arguments and then straight pass them into the main function of the script which is usually in the same name as the file for example here it could be `create_training_data_from_prodigy(args.prodigy_data_dir)`.

This makes the __main__ part smaller the script easier to test.